src/libfetchers/git.cc

```diff
@@ -501,6 +507,9 @@ struct GitInputScheme : InputScheme
 
         input->submodules = maybeGetBoolAttr(attrs, "submodules").value_or(false);
 
+        if (maybeGetBoolAttr(attrs, "gitIngestion").value_or(false) || input->treeHash)
```

Comment:
```suggestion
        if (maybeGetBoolAttr(attrs, "gitIngestion").value_or(input->treeHash))
```
or if that doesn't work
```suggestion
        if (maybeGetBoolAttr(attrs, "gitIngestion").value_or((bool) input->treeHash))
```